aiida/cmdline/commands/cmd_calcjob.py

```diff
@@ -229,7 +230,10 @@ def calcjob_outputls(calcjob, path, color):
 @options.COMPUTERS(help='include only calcjobs that were ran on these computers')
 @options.FORCE()
 @options.EXIT_STATUS()
-def calcjob_cleanworkdir(calcjobs, past_days, older_than, computers, force, exit_status):
+@click.option(
+    '--include-cleaned', is_flag=True, default=False, help='Included calcjobs that have been cleaned previously'
+)
+def calcjob_cleanworkdir(calcjobs, past_days, older_than, computers, force, exit_status, include_cleaned):
```

Comment:
Can we just set this internally rather than an option to set that always ignores the 'cleaned' calcjobs?  Do you see any downside of it?

Reply:
I can see another advantage of adding an extra attribute to the calcjob that has been cleaned, is that it will be easy to turn off the caching for the process specifically.

Reply:
My worry was that for very large database trying to find the `cleaned` flag may be slow? Otherwise, I agree we should skip the cleaned ones automatically. 

> I can see another advantage of adding an extra attribute to the calcjob that has been cleaned, is that it will be easy to turn off the caching for the process specifically.

The flag should not affect caching at all, it is fine to cache a calcultion containing output remote folder that is cleaned. But runing a new calculation taking a cleaned remote folder as input would not make any sense.

Reply:
>  >I can see another advantage of adding an extra attribute to the calcjob that has been cleaned, is that it will be easy to turn off the caching for the process specifically.
>
>The flag should not affect caching at all, it is fine to cache a calcultion containing output remote folder that is cleaned. But runing a new calculation taking a cleaned remote folder as input would not make any sense.

I was to going to say since the clean workdir may have some downside influence on the caching mechenism, like in the workflow the previous Calcjob has a RemoteData output which used for the subsequent CalcJob in the workflow but the RemoteData workdir is cleaned. (For reference, check discussion https://github.com/aiidateam/aiida-core/issues/5178). Then with you implementation here you add a `workdir_cleaned` to the extra attributes, I then can check these attributes and control the caching (by the new feature implemented https://github.com/aiidateam/aiida-core/issues/5189) for this specific type of 'cleaned nodes' and turn off the caching for them.

Reply:
To solve #5178, what you really need is have AiiDA to work out a sequence of non-cached calculations to run with a certain missing `RemoteData` end point, which would be quite complex to implement...
 
Suppose you have a workchain that runs fource calculations, *A*, *B*, *C* and *D* in a sequence, each using the `RemoteData` from the previous one.

*A* -> *RemoteA* -> *B* -> *RemoteB* -> *C* -> *RemoteC* -> *D*

If *D* crashes, restarting the workflow with caching on should fast-forward to running *D*, this is done by caching *A*, followed by *B*, then by *C* in a seqeuence. The remote folder of *A* and *B* may be cleaned and it should not affect having *C* cached. If the `RemoteFolder` of *C* is cleaned then obviously the restart of *D* should not proceed, but this is not related to caching mechanism. 

Instead, attempt to run *D* should just give an error, and one has to run the workchain without caching from scratch. However, this does not stop *A*, *B*, *C* being valid caches, as some calculations may not use `RemoteData` as inputs, and there is no ways to know how future workchain will use existing nodes. 

If we make a cleaned `RemoteFolder` invalidating the cache, then the chain would be breakon in the first step, which is a bit of overkill.